# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSScript.h
# Commit: 2a0aacbd11de
# Full Hash: 2a0aacbd11dee6907445c902aa088ab4b9492d6b
# Author: caroline <cullen.caroline@gmail.com>
# Date: 2019-12-10 21:29:05
# Regressor Bug: 1588861
# File Overlap Count: 1
# Description:
#   Bug 1588861 - Part 3: Add support for xdr encoding modules into xdr cache. r=iain,tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54158
# ==============================================================================

diff -r 639290e9ff22 -r 2a0aacbd11de js/src/vm/JSScript.h
--- a/js/src/vm/JSScript.h	Tue Dec 10 02:52:30 2019 +0000
+++ b/js/src/vm/JSScript.h	Tue Dec 10 02:52:37 2019 +0000
@@ -1592,7 +1592,7 @@
                                     js::HandleScript script,
                                     js::HandleScriptSourceObject sourceObject,
                                     js::HandleScope scriptEnclosingScope,
-                                    js::HandleFunction fun);
+                                    js::HandleObject funOrMod);
 
   // Clone src script data into dst script.
   static bool Clone(JSContext* cx, js::HandleScript src, js::HandleScript dst,
@@ -2158,8 +2158,8 @@
  */
 template <XDRMode mode>
 XDRResult XDRScript(XDRState<mode>* xdr, HandleScope enclosingScope,
-                    HandleScriptSourceObject sourceObject, HandleFunction fun,
-                    MutableHandleScript scriptp);
+                    HandleScriptSourceObject sourceObject,
+                    HandleObject funOrMod, MutableHandleScript scriptp);
 
 template <XDRMode mode>
 XDRResult XDRLazyScript(XDRState<mode>* xdr, HandleScope enclosingScope,
@@ -2576,7 +2576,7 @@
   friend js::XDRResult js::XDRScript(js::XDRState<mode>* xdr,
                                      js::HandleScope enclosingScope,
                                      js::HandleScriptSourceObject sourceObject,
-                                     js::HandleFunction fun,
+                                     js::HandleObject funOrMod,
                                      js::MutableHandleScript scriptp);
 
   template <js::XDRMode mode>
@@ -2599,7 +2599,7 @@
   friend js::XDRResult js::PrivateScriptData::XDR(
       js::XDRState<mode>* xdr, js::HandleScript script,
       js::HandleScriptSourceObject sourceObject,
-      js::HandleScope scriptEnclosingScope, js::HandleFunction fun);
+      js::HandleScope scriptEnclosingScope, js::HandleObject funOrMod);
 
   friend bool js::PrivateScriptData::Clone(
       JSContext* cx, js::HandleScript src, js::HandleScript dst,